src/test/jtx/Env_test.cpp

```diff
@@ -49,13 +49,6 @@ class Env_test : public beast::unit_test::suite
     testAccount()
     {
         using namespace jtx;
-        {
-            Account a;
-            Account b(a);
-            a = b;
-            a = std::move(b);
-            Account c(std::move(a));
-        }
```

Comment:
I'm not a huge fan of the `static_asserts` being sprinkled in PublicKey.h and jtx/Account.h.  I've actually done similar stuff when I was designing new classes and I wanted to be confident I had the right semantics.  But it was usually during initial design.  I've also sometimes added similar `static_assert`s if I was modifying a class that I suspected I didn't understand well.

However both `PublicKey` and `jtx::Account` are well exercised and wrung out classes.  And we're not modifying them.  Adding the `static_assert`s there does not really improve our confidence that the classes are doing the right thing.  We already have pretty good confidence in that.

If we move our focus to the (above) code in the braces, that code is just verifying that all of the usual constructors and assignment operators compile and don't crash if they execute.  So we could hypothetically replace them with `static_assert`s like this:
```
        static_assert(std::is_default_constructible_v<Account>);
        static_assert(std::is_copy_constructible_v<Account>);
        static_assert(std::is_move_constructible_v<Account>);
        static_assert(std::is_copy_assignable_v<Account>);
        static_assert(std::is_move_assignable_v<Account>);
```
But there's a (small) problem here as well.  The `static_assert`s only show that the compiler is fine with calling those special member functions.  What the `static_assert`s _don't_ demonstrate is whether calling those special member functions crashes.  So by switching to the `static_assert`s we are actually reducing our test coverage.

Now, in truth, at this point in time we're pretty well convinced that all these special member functions:

1. Are callable,
2. Don't crash,  and
3. Actually do what they are expected to do (which, incidentally, the code in braces does not verify).

So, while the changes in this pull request are interesting, I don't know that they actually bring a lot of value to the code base.

I'm open to discussion if you have questions or disagree.